packages/react-native/ReactCommon/react/performance/timeline/PerformanceEntryBuffer.h

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include "PerformanceEntry.h"
+
+namespace facebook::react {
+
+// Default duration threshold for reporting performance entries (0 means "report
+// all")
+constexpr double DEFAULT_DURATION_THRESHOLD = 0.0;
+
+/**
+ * Abstract performance entry buffer with reporting flags.
+ * Subtypes differ on how entries are stored.
+ */
+class PerformanceEntryBuffer {
+public:
+  double durationThreshold{DEFAULT_DURATION_THRESHOLD};
+  size_t droppedEntriesCount{0};
+
+  explicit PerformanceEntryBuffer() = default;
+  virtual ~PerformanceEntryBuffer() = default;
+
+  virtual void add(const PerformanceEntry& entry) = 0;
+  virtual void consume(std::vector<PerformanceEntry>& target) = 0;
+  virtual size_t pendingMessagesCount() const = 0;
```

Comment:
Nit on naming: we don't use the concept of "message" anywhere else here, so perhaps we should just call it `getCount()`.

On the other hand, we're not using this method anywhere so maybe we can just remove it.